File: Problem54_DeleteWithoutHeadPointer.java
// Time Complexity : O(1)
// Space Complexity : O(1)
class Solution
{
    void deleteNode(Node del)
    {
         // Your code here
        Node temp = del.next;
        del.data = temp.data;
        del.next = temp.next;
        temp = null;
    }
}

File: Problem-54 Delete without head pointer .py
# Delete without head pointer 
# https://practice.geeksforgeeks.org/problems/delete-without-head-pointer/1

# Time Complexity: O(1)
# Space Complexiety: O(1)

# if ptr is the node given and it points to the node that is to be deleted.
if ptr.next:
    ptr.val = ptr.next.val
    ptr.next = ptr.next.next
else:
    ptr = None